File: backend/dataset/levelOrder/levelorder_15.py
from collections import deque
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        queue=deque()
        queue.append(root)
        if not root: return []
        result = [[root.val]]
        n1 = []
        n2 = [1]
        while queue:
            n1=[]
            for _ in range(len(n2)):
                current=queue.popleft()
                if(current.left): 
                    queue.append(current.left)
                    n1.append(current.left.val)
                if(current.right): 
                    queue.append(current.right)
                    n1.append(current.right.val)
            n2=n1
            if(len(n2)): result.append(n2)
        return result

File: backend/CDataset/levelOrder/levelOrder_15.cpp
class Solution {
public:
    #include<vector>

    class TreeNode{
        public:
        TreeNode* left;
        TreeNode* right;
        int val;
    };
vector<vector<int>> levelOrder(TreeNode* root) {
       queue<TreeNode*> Q;
        vector<vector<int>> res;
        if(root == nullptr) return res;
        Q.push(root);
        while(Q.size() != 0) {
            int size = Q.size();
            vector<int> v;
            for(int i = 0; i < size; i++) {
                 TreeNode* t = Q.front();
                 Q.pop();
                 v.push_back(t->val);
                if(t->left != nullptr) 
                    Q.push(t->left);
                if(t->right != nullptr)
                    Q.push(t->right);
            }
            res.push_back(v);
        }
        return res;
    }
};